Feature output all possible outputs

hsteinacker-siz opened the pull request:
- "output_keys" and "output_plot" in the input file can now be "all", "nothing" or a list of entries as before for custom outputs of CSV and lineplot
- output_values() of all components was changed to not only return the available channels, but also the corresponding media
- restructured parts of "Resie.jl" for better readability (could probabely be further improved...)
 
Minor version bump needs to be done (backwards compatibility with old input files is given).

eott-siz commented on src/Resie.jl:
```diff
@@ -62,69 +62,91 @@ function run_simulation(project_config::Dict{AbstractString,Any})
     )
     EnergySystems.set_timestep(parameters["time_step_seconds"])
 
```
The function run_simulation was already a bit long before the changes in this PR, but I think it might be time to move the functionality for gathering outputs into smaller functions to improve readability.

hsteinacker-siz replied:
I thought the same...

eott-siz commented on src/Resie.jl:
```diff
@@ -62,69 +62,91 @@ function run_simulation(project_config::Dict{AbstractString,Any})
-        project_config["io_settings"]["output_plot"] === "nothing"
-    )
+    # read in flags, if all outputs should be printed and/or plotted
+    plot_all_outputs_bool = false
```
Small nitpick, but generally variable names should not (need to) contain their type. For boolean variables one convention is to use a "is_", has_" or "do_" prefix.

hsteinacker-siz replied:
Your right, I will change that!

eott-siz reviewed (commented):
Functionality-wise this looks good. Before we merge, we can use the opportunity to improve readability of the code handling output.

hsteinacker-siz commented:
@eott-siz 
I renamed the variables and moved many functions from Resie.jl to file_output.jl for better readability. Should be merge now, or do you have any more suggestions?
